### PR TITLE
add tests for textarea with field method #220

### LINKED ISSUE #220: PROBLEM WITH FILLING TEXTAREA
abhipata...@gmail.com reported on Feb 14, 2013 

<pre>
<b>What steps will reproduce the problem?</b>
<b>1.</b>
The text area is:
&lt;textarea rows=&quot;5&quot; cols=&quot;60&quot; id=&quot;p_input&quot; name=&quot;p_input&quot;&gt;&lt;/textarea&gt;
<b>2.</b>
I m filling this textarea using the following code
my $mech = WWW::Mechanize-&gt;new();
$mech-&gt;timeout(5000);
$mech-&gt;agent(&#39;Mozilla/5.0&#39;);
$mech-&gt;proxy([&#39;http&#39;], &#39;http://202.41.70.84:6588&#39;);

	$url=&quot;<a href="http://pridb.gdcb.iastate.edu/RPISeq/index.html&quot;" rel="nofollow">http://pridb.gdcb.iastate.edu/RPISeq/index.html&quot;;</a>
	$mech-&gt;get($url);
	die $mech -&gt;res-&gt;status_line unless $mech -&gt;success ;
	$mech-&gt;form_name( &#39;myform&#39; ) or die (&quot;CANT FIND&quot;);

	
	my @in = $mech-&gt;find_all_inputs(
        type       =&gt; &#39;textarea&#39;,
        name_regex =&gt; qr/input/,
    );								
	print @in;
	$one=&quot;p_input&quot;;
	$two=&quot;r_input&quot;;
		$mech-&gt; field($one,&quot;MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM&quot;) or die (&quot;CF&quot;);
		$mech-&gt; field($two,&quot;AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA&quot;);
								    							   		$mech-&gt;submit();

3. This is not working and i am getting the death &#39;C F&#39;

<b>What is the expected output? What do you see instead?</b>
I get a death instead out the html page output

<b>What version of the product are you using? On what operating system?</b>
I am using this very mechanize on windows 7

<b>Please provide any additional information below.</b>

</pre>

### Details

Imported from [Google Code issue 243](http://code.google.com/p/www-mechanize/issues/detail?id=243) via [archive](https://web.archive.org/web/20150227111857/http://code.google.com/p/www-mechanize/issues/detail?id=243)

* **Type**:      Defect
* **Date**:      Feb 14, 2013
* **Reporter**:  abhipata...@gmail.com
* **Owner**:     ----
* **Priority**:  Medium
* **Status**:    New

### PR DESCRIPTION
**Pull request by simbabque**
I couldn't reproduce this issue, but I've added tests.

Closes #220.

### CODE REVIEW

File: t/field.t [hunk 1]
```diff
@@ -29,6 +29,10 @@ $mech->set_visible( [ radio => 'wongo!' ], 'boingo' );
 is( $mech->value( 'wango' ), 'wongo!', 'wango changed' );
 is( $mech->value( 'dingo', 2 ), 'boingo', 'dingo changed' );
 
+ok( ! $mech->value( 'textarea_name' ), 'textarea is empty' );
+$mech->field( 'textarea_name' => 'foobar' );
+is( $mech->value( 'textarea_name' ), 'foobar', 'textarea is empty' );
```
Comment (oalders) [inline t/field.t:34]:
```suggestion
is( $mech->value( 'textarea_name' ), 'foobar', 'textarea has been populated' );
```

**Review by oalders — CHANGES_REQUESTED** [review]
Looks great. I think one of the test labels was a victim of copy/paste?

**Comment by simbabque** [comment]
> Looks great. I think one of the test labels was a victim of copy/paste?

Yes, it was. Thanks for spotting that. I've pushed a new version.

**Review by oalders — APPROVED** [review]
🏅

**Pull request merged**